tests/test_location_pi.py

```diff
@@ -394,7 +458,69 @@ def test14_get_stock_drift_new_package_token_multiple_products(self):
         self.assertEqual(package_ids[0], package_ids[1],
                          "Did not use the same package")
 
-    def test15_get_stock_drift_no_package_token_with_lot(self):
+    def test17_get_stock_drift_two_new_package_token_multiple_products(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains two 'NEWPACKAGE' packages; multiple
+        product requests, with the different package names (multiple
+        packages will be created).
+        """
+        self.create_quant(self.apple.id, self.test_location_01.id, 5)
+        self.create_quant(self.banana.id, self.test_location_01.id, 6)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": "test17_NEWPACKAGE_one",
+                     "quantity": 25},
+                    {"product_id": self.banana.id,
+                     "package_name": "test17_NEWPACKAGE_two",
+                     "quantity": 26}]
+
+        stock_drift = self.test_location_01._get_stock_drift(adjs_req)
+
+        self.assertEqual(len(stock_drift), 2,
+                         "Does not contain two stock drift entries")
+
+        package_ids = []
+
+        for stock_info, qty in stock_drift.items():
+            if stock_info.product_id == self.apple.id:
+                self.assertEqual(qty, 25, "Wrong quantity")
+            elif stock_info.product_id == self.banana.id:
+                self.assertEqual(qty, 26, "Wrong quantity")
+            else:
+                self.fail("Unexpected product id")
+
+            package_ids.append(stock_info.package_id)
+
+        self.assertNotEqual(package_ids[0], package_ids[1],
+                            "Did use the same package")
+
+    def test18_get_stock_drift_new_package_token_more_products_one_adj(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains a single existent package; multiple
+        products are within such package, but only one product needs
+        to be adjusted.
+        """
+        self.create_quant(self.apple.id, self.test_location_01.id, 65,
+                          package_id=self.package_one.id)
+        self.create_quant(self.banana.id, self.test_location_01.id, 66,
+                          package_id=self.package_one.id)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": self.package_one.name,
+                     "quantity": 65}]
```

Comment:
We already have 65 apples, is this intended?

Reply:
Oh, that should be different than what's in the quant for readability, but it doesn't affect the test.

Reply:
I agree, but it might add confusion